### ISSUE TITLE
Minigalaxy launch size is too large for Steam Deck screen, would be cool if it could be adjusted to work!

### ISSUE THREAD

**Issue by ell1e** [sharkwouter/minigalaxy#604]
I just tried Minigalaxy on the Steam Deck since that would be kind of the ultimate open-source move, using a nice DRM-free platform with an open-source client. Sadly, the minimum size of the Minigalaxy main window seems to be considerably too large:

![Screenshot_20240703_164328](https://github.com/sharkwouter/minigalaxy/assets/64124388/0a544b26-86a5-4635-8bd3-ce7c93eb5c8d)

But bafflingly, after tabbing in and out of it for a while, it suddenly resized to fit on the screen. So it *can fit* but right after launching it, it just doesn't. If it was possible to fix this at some point that would be really cool, since minigalaxy on the Steam Deck is just a fun idea!

I tested this with **Minigalaxy 1.3.0 from Flathub**

**Comment by sharkwouter** [comment]
I guess it could be make to default to maximized view instead of a set width and height.

**Comment by ell1e** [comment]
How about checking the size of the smallest screen connected and if it doesn't fit, launch maximized? That might do the job while not bothering anybody with larger screens.

**Comment by sharkwouter** [comment]
I think that should be possible. I think GLib or Gtk will have some function for that.

**Comment by ell1e** [comment]
It seems like this should perhaps be handled by KDE/KWin instead, upstream discussion is here: https://bugs.kde.org/show_bug.cgi?id=498221 I'll let you know how it resolves.

**Comment by sharkwouter** [comment]
Interesting, thanks for linking this.

**Comment by ell1e** [comment]
Ongoing discussion of [.set_default_size()](https://docs.gtk.org/gtk4/method.Window.set_default_size.html) running into this problem [happening over at GTK+](https://gitlab.gnome.org/GNOME/gtk/-/issues/7253). I'll let you know how it goes!

**Comment by sharkwouter** [comment]
Doesn't gtk have scaling support? Maybe we could do a 50% scale.

**Comment by ell1e** [comment]
Then it would probably no longer be possible to tap items on the Steam Deck because they would be too small. I'm guessing if GTK+ confirms this is a GTK+ bug, then the best solution might be to have them fix it and leave minigalaxy as it is.

**Comment by ell1e** [comment]
Currently, my personal impression was that the GTK+ devs don't seem to know themselves if GTK+ should handle this or the app should, given GTK 4 does this right but somehow they were arguing against it, and GTK 3 doesn't do it right. Qt and SDL3 seem to enforce this properly now, I guess for GTK it's up in the air and might still be worth fixing in minigalaxy for now, if it's not too hard to do. (I imagine at least with multiple screens it might be difficult though, I've heard with Wayland it's not always predictable which screen a window will even open on.)

**Comment by GB609** [comment]
Just for further testing: Which size in pixels does the Steamdeck have?
If it works the same as smartphones, it's not the pixel amount itself that is too small, but the display manager scaling stuff because it would be too small to see otherwise.

**Comment by sharkwouter** [comment]
The Steam deck's screen is 1280 by 800 pixels.

Honestly, we should probably switch to GTK 4. That would also resolve the issue where the login page is not shown on some systems.

**Comment by GB609** [comment]
Good idea for the next release. But let me finish my download ui first. Afterwards we could do a ui migration step. It'd be too confusing to change in the middle of the development now.

**Comment by ell1e** [comment]
The problem is in part the pixel density, I think the logical size of the screen is something like 640 x 400 with the install I use. But the main thing that matters in the end is that if manually resized, minigalaxy fits and works perfectly fine.
